Diagnostic/diagnostic.py

```diff
@@ -642,6 +649,19 @@ def start_mdsd():
             copy_env['MDSD_http_proxy'] = proxy_config
 
     xml_file = os.path.join(WorkDir, './xmlCfg.xml')
+
+    # We now validate the config and proceed only when it succeeds.
+    config_validate_cmd = '{0} -v -c {1}'.format(os.path.join(MdsdFolder, "mdsd"), xml_file)
+    config_validate_cmd_status, config_validate_cmd_msg = RunGetOutput(config_validate_cmd)
+    if config_validate_cmd_status is not 0:
+        # Invalid config. Log error and report success.
+        message = "Invalid mdsd config given. Can't enable. Still considered successful. Config validation result: "\
```

Comment:
It means the extension install/enable is considered successful (as it's a user error), not meaning that mdsd is OK with this invalid configuration. This invalid config issue shouldn't penalize the extension install/failure success rate. I'll update the comment so that it's clearer.